### PR TITLE
Editorial: Make [[HasProperty]] consistent with other internal methods

### LINKED ISSUE #394: Ordinary* methods inconsistent in invoking [[*]] vs Ordinary* in their steps
`OrdinaryHasProperty` invokes `OrdinaryGetOwnProperty` directly, not `[[GetOwnProperty]]`.

The other `Ordinary*` methods we introduced invoke `[[*]]` rather than `Ordinary*` in their algorithms. The former is more useful for reuse by the `WindowProxy` and `Location` exotic objects, but I can imagine the latter being important for some cases too. Is there a reason `OrdinaryHasProperty` behaves the way it does?

Even if `OrdinaryHasProperty` remains the way it is, can I be sure that we will not change `OrdinaryGet` et al from invoking `[[*]]` to invoking `Ordinary*`?

### PR DESCRIPTION
**Pull request by annevk**
Fixes #394.

### CODE REVIEW

File: spec.html [hunk 1]
```diff
@@ -7610,6 +7591,20 @@
...
+        <h1>[[HasProperty]] ( _P_ )</h1>
+        <p>The [[HasProperty]] internal method of an arguments exotic object when called with property key _P_, performs the following steps:</p>
+        <emu-alg>
+          1. Let _args_ be the arguments object.
+          1. If _P_ is `"caller"`, then
+            1. Let _desc_ be OrdinaryGetOwnProperty(_args_, _P_).
+            1. If IsDataDescriptor(_desc_) is *true*, return *true*.
+          1. Return OrdinaryHasProperty(_args_, _P_).
+
+          NOTE: If an implementation does not provide a built-in caller property for argument exotic objects then step 2 of this algorithm must be skipped.
```
Comment (bterlson) [inline spec.html]: This should be in a separate emu-note below the alg.

Comment (annevk) [inline spec.html]: Actually, it should just be a paragraph since it's normative. I should have looked at the other occurrence.

**Comment by annevk** [comment]
This should be good now. I also added ?/! as appropriate.

**Comment by bterlson** [comment]
Ok, seems good.

**Comment by bterlson** [comment]
Committed as 6ef0ee3.

**Pull request closed without merging**